presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSourcePartitionedScheduler.java

```diff
@@ -156,9 +159,10 @@ public ScheduleResult schedule()
         // schedule a task on every node in the distribution
         List<RemoteTask> newTasks = ImmutableList.of();
         if (!scheduledTasks) {
-            OptionalInt totalPartitions = OptionalInt.of(partitioning.getPartitionToNode().size());
-            newTasks = partitioning.getPartitionToNode().entrySet().stream()
-                    .map(entry -> stage.scheduleTask(entry.getValue(), entry.getKey(), totalPartitions))
+            OptionalInt totalPartitions = OptionalInt.of(nodes.size());
+            newTasks = Streams.mapWithIndex(
+                    nodes.stream(),
+                    (node, taskId) -> stage.scheduleTask(node, toIntExact(taskId), totalPartitions))
```

Comment:
The variable name `taskId` is a bit misleading here. Use `i` instead.

In addition, I believe this would be easier to read if written as a loop. But I don't feel strongly.

```
            ImmutableList.Builder<RemoteTask> newTasksBuilder = ImmutableList.builder();
            for (int i = 0; i < nodes.size(); i++) {
                stage.scheduleTask(nodes.get(i), i, totalPartitions);
            }
            newTasks = newTasksBuilder.build();
```